conans/client/tools/files.py

```diff
@@ -224,17 +225,34 @@ def _manage_text_not_found(search, file_path, strict, function_name, output):
         return False
 
 
+@contextmanager
+def add_permissions(file_path, permissions):
+    if os.path.isfile(file_path):
+        saved_permissions = os.stat(file_path).st_mode
+        if saved_permissions & permissions == permissions:
+            yield
+            return
+        try:
+            os.chmod(file_path, saved_permissions | permissions)
+            yield
+        finally:
+            os.chmod(file_path, saved_permissions)
+    else:
+        yield
+
 def replace_in_file(file_path, search, replace, strict=True, output=None, encoding=None):
     output = default_output(output, 'conans.client.tools.files.replace_in_file')
 
     encoding_in = encoding or "auto"
     encoding_out = encoding or "utf-8"
-    content = load(file_path, encoding=encoding_in)
+    with add_permissions(file_path, stat.S_IREAD):
+        content = load(file_path, encoding=encoding_in)
     if -1 == content.find(search):
         _manage_text_not_found(search, file_path, strict, "replace_in_file", output=output)
     content = content.replace(search, replace)
     content = content.encode(encoding_out)
-    save(file_path, content, only_if_modified=False, encoding=encoding_out)
+    with add_permissions(file_path, stat.S_IWRITE):
```

Comment:
I am not fully convinced about this approach, something seems not completely right, like this module shouldn't be responsible of doing this. If the user do a ``save()`` it will fail due to permissions, but ``replace_in_file()`` should succeed?

Asking @jgsogo for a second opinion.

Reply:
in the first attempt I put it to the `save` method, but it's used extensively in many other tools and internal functions, as result it caused many different tests to fail (e.g. `test_export_read_only`). so I decided to isolate it, and limit changes just to `replace_in_file`/`replace_path_in_file` family of functions.